mmv1/third_party/terraform/services/netapp/resource_netapp_volume_test.go

```diff
@@ -0,0 +1,344 @@
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
+package netapp_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccNetappVolume_volumeBasicExample_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"network_name":  acctest.BootstrapSharedServiceNetworkingConnection(t, "gcnv-network-config-1", acctest.ServiceNetworkWithParentService("netapp.servicenetworking.goog")),
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckNetappVolumeDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNetappVolume_volumeBasicExample_basic(context),
+			},
+			{
+				ResourceName:            "google_netapp_volume.test_volume",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "name", "labels", "terraform_labels"},
+			}, {
+				Config: testAccNetappVolume_volumeBasicExample_full(context),
+			},
+			{
+				ResourceName:            "google_netapp_volume.test_volume",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "name", "labels", "terraform_labels"},
+			},
+			{
+				Config: testAccNetappVolume_volumeBasicExample_update(context),
+			},
+			{
+				ResourceName:            "google_netapp_volume.test_volume",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "name", "labels", "terraform_labels"},
+			},
+			{
+				Config: testAccNetappVolume_volumeBasicExample_updatesnapshot(context),
+			},
+			{
+				ResourceName:            "google_netapp_volume.test_volume",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "name", "labels", "terraform_labels"},
+			},
+		},
+	})
+}
+
+func testAccNetappVolume_volumeBasicExample_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_netapp_storage_pool" "default" {
+	name = "tf-test-test-pool%{random_suffix}"
+	location = "us-central1"
+	service_level = "PREMIUM"
+	capacity_gib = "2048"
+	network = data.google_compute_network.default.id
+}
+
+resource "google_netapp_volume" "test_volume" {
+	location = "us-central1"
+	name = "tf-test-test-volume%{random_suffix}"
+	capacity_gib = "100"
+	share_name = "tf-test-test-volume%{random_suffix}"
+	storage_pool = google_netapp_storage_pool.default.name
+	protocols = ["NFSV3"]
+	smb_settings = []
+	unix_permissions = "0770"
+	labels = {}
```

Comment:
can we have `smb_settings` and `labels` set to values other than empty values in some of the test steps?

Reply:
Labels are covered, see line 139.
smb_settings is unfortunately very complex and an are we don't know how to test properly in the test environment. To do anything related to SMB or Kerberos, we need an Active Directory available, since that's an external dependency to create such environments. Since we don't have an AD in the test environment, we cannot run automated tests. ManagedAD won't work do to VPC peering issues.
I did some basic testing of the provider in my lab environment, testing SMB volume creation and smb_settings.

Any idea how to resolve that in the test harness (=any way to provide an Active Directory domain controller?)